File: src/main/java/j/examples/rules/guarding/PdfExtensionFileRule.java
package j.examples.rules.guarding;

import j.examples.validation.ValidationException;

import java.io.File;

public class PdfExtensionFileRule {

    public void validate(File file) {
        String fileExtension = file.getName().substring(file.getName().lastIndexOf('.') + 1);

        if (!fileExtension.equals("pdf")) {
            throw new ValidationException(
                "The file %s is not a .pdf file".formatted(file)
            );
        }
    }
}

File: src/main/kotlin/k/examples/rules/guarding/PdfExtensionFileRule.kt
package k.examples.rules.guarding

import k.examples.validation.ValidationException
import java.io.File

class PdfExtensionFileRule {
    fun validate(file: File) {
        if (file.extension != "pdf") {
            throw ValidationException(
                "The file ${file} is not a .pdf file"
            )
        }
    }
}